rkt/api_service.go

```diff
@@ -636,8 +639,54 @@ func (s *v1AlphaAPIServer) InspectImage(ctx context.Context, request *v1alpha.In
 	return &v1alpha.InspectImageResponse{Image: image}, nil
 }
 
+type LogsStreamWriter struct {
+	server v1alpha.PublicAPI_GetLogsServer
+}
+
+func (sw LogsStreamWriter) Write(b []byte) (int, error) {
+	if err := sw.server.SendMsg(b); err != nil {
+		return 0, err
+	}
+	return len(b), nil
+}
+
 func (s *v1AlphaAPIServer) GetLogs(request *v1alpha.GetLogsRequest, server v1alpha.PublicAPI_GetLogsServer) error {
-	return fmt.Errorf("not implemented yet")
+	path := fmt.Sprintf("/var/lib/rkt/pods/run/%s/stage1/rootfs/var/log/journal/", request.PodId)
+	jconf := sdjournal.JournalReaderConfig{
+		Path: path,
+	}
+	if request.AppName != "" {
+		jconf.Matches = []sdjournal.Match{
+			{
+				Field: sdjournal.SD_JOURNAL_FIELD_SYSTEMD_UNIT,
+				Value: fmt.Sprintf("%s.service", request.AppName),
+			},
+		}
+	}
+	if request.SinceTime != 0 {
+		t := time.Unix(request.SinceTime, 0)
+		jconf.Since = time.Since(t)
+	}
+	if request.Lines != 0 {
+		jconf.NumFromTail = uint64(request.Lines)
+	}
+
+	jr, err := sdjournal.NewJournalReader(jconf)
+	if err != nil {
+		return err
+	}
+
+	if request.Follow {
+		return jr.Follow(nil, LogsStreamWriter{server: server})
+	} else {
+		data, err := ioutil.ReadAll(jr)
```

Comment:
If the journal is huge and the caller didn't use `request.Lines`, this might use too much memory. Would the following work?
```
_, err := io.Copy(LogsStreamWriter{server: server}, jr)
```